src/java.base/share/classes/java/nio/channels/Channels.java

```diff
@@ -260,8 +260,9 @@ public void close() throws IOException {
      * Constructs a channel that reads bytes from the given stream.
      *
      * <p> The resulting channel will not be buffered; it will simply redirect
-     * its I/O operations to the given stream.  Closing the channel will in
-     * turn cause the stream to be closed.  </p>
+     * its I/O operations to the given stream. Reading from the resulting
+     * channel will read from the input stream hence might block. Closing the
```

Comment:
A suggestion is to replace "will read from the input stream hence might block" with "will read from the input stream and thus block until input is available or end of file is reached", would that work?

Reply:
That is indeed more specific.